### PR TITLE
Replace icons with Toucan Icons

### LINKED ISSUE #49: Replace icons with toucan icons
At the moment we are using placeholder icons, but we should update all existing placeholders with the real Toucan icons.

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR replaces the placeholder icons with Toucan icons.

Closes #49 

---

## 🔬 How to Test

- Visit https://8d11545e.ember-toucan-core.pages.dev/docs/components/checkbox-field
  - Check the checkbox and view the new checkmark
  - Go to https://8d11545e.ember-toucan-core.pages.dev/docs/components/checkbox-field#checkboxfield-with-isindeterminate and view the indeterminate icon
- Visit https://8d11545e.ember-toucan-core.pages.dev/docs/components/file-input-field
  - View the new upload icon
  - View the new trash icon
- Visit https://8d11545e.ember-toucan-core.pages.dev/docs/components/textarea-field#textareafield-with-label-and-error
  - View the new error icon

---

## 📸 Images/Videos of Functionality

| BEFORE  | AFTER |
| ------------- | ------------- |
| Screenshot 2023-04-20 at 11 57 36 AM  | Screenshot 2023-04-20 at 11 57 57 AM  |
| Screenshot 2023-04-20 at 11 58 36 AM  | Screenshot 2023-04-20 at 11 59 00 AM  |
| Screenshot 2023-04-20 at 11 59 24 AM | Screenshot 2023-04-20 at 11 59 58 AM |

### CODE REVIEW

File: packages/ember-toucan-core/src/-private/components/error.hbs
Comment (ynotdraw) [inline packages/ember-toucan-core/src/-private/components/error.hbs:1]: We will probably want an Icon component at some point as we re-use some of these, but for the time being this (being inline) is probably fine?

File: packages/ember-toucan-core/src/-private/components/error.hbs [hunk 1]
```diff
@@ -3,21 +3,22 @@
...
 >
-  {{! TODO: https://github.com/CrowdStrike/ember-toucan-core/issues/49 - Placeholder icon }}
   <svg
-    class="mr-1 h-3 w-3"
-    xmlns="http://www.w3.org/2000/svg"
-    width="24"
-    height="24"
-    stroke="currentColor"
-    viewBox="0 0 24 24"
+    aria-hidden="true"
```
Comment (nicolechung) [inline packages/ember-toucan-core/src/-private/components/error.hbs:7]: Should `aria-hidden` true be on all icons? TIL...

**Review by nicolechung — APPROVED** [review]

File: packages/ember-toucan-core/src/-private/components/error.hbs (hunk 1 above)
Comment (ynotdraw) [reply packages/ember-toucan-core/src/-private/components/error.hbs:7]: Yeah! I think it's generally a recommended practice to remove it from the Accessibility Tree as they're only presentational additions.  We have some existing code doing this as well.  [This stack overflow](https://stackoverflow.com/questions/61048356/why-do-we-use-aria-hidden-with-icons) explains it a bit better (they use `<i` instead)

**Comment by ynotdraw** [comment]
Once this gets merged, I'll cut a release of `ember-toucan-form` 🎉

**Pull request merged**